src/main/webapp/app/course/learning-paths/participate/learning-path-container.component.ts

```diff
@@ -57,8 +57,16 @@ export class LearningPathContainerComponent implements OnInit {
                 this.courseId = params['courseId'];
             });
         }
-        this.learningPathService.getLearningPathId(this.courseId).subscribe((learningPathIdResponse) => {
-            this.learningPathId = learningPathIdResponse.body!;
+        this.learningPathService.getLearningPathId(this.courseId).subscribe({
+            next: (learningPathIdResponse) => {
+                this.learningPathId = learningPathIdResponse.body!;
+            },
+            error: () => {
+                // if the learning path does not exist, we do need to create it
```

Comment:
Does it really make sense to generate the learning path for every error that could occur? I think the code would be more robust if you check that the server returned an appropriate error message, e.g. a 404